lib/puppet/data_providers.rb

```diff
@@ -25,6 +27,9 @@ def self.lookup_in_module(name, lookup_invocation, merge)
   end
 
   def self.lookup_adapter(lookup_invocation)
+    unless Puppet[:strict] == :off
+      Puppet.deprecation_warning('The method Puppet::DataProviders.lookup_adapter is deprecated and will be removed in the next major release of Puppet.')
```

Comment:
Does this use the "warn once" logic? Otherwise there is the risk of getting a lot of deprecation warnings in the log.